oqsprov/oqsprov_keys.c

```diff
@@ -763,10 +764,7 @@ OQSX_KEY *oqsx_key_new(OSSL_LIB_CTX *libctx, char *oqs_name, char *tls_name,
 
 #ifdef OQS_PROVIDER_NOATOMIC
     ret->lock = CRYPTO_THREAD_lock_new();
-    if (ret->lock == NULL) {
-        OPENSSL_free(ret);
-        goto err;
-    }
+    ON_ERR_GOTO(!ret->lock, err);
```

Comment:
> Do you think we should also follow this rule for oqs-provider?

We could do that -- but I'd rather have that as a separate issue/PR. IMO it would also entail more work, e.g., no use of C++ style comments.... and we've got plenty of those.